core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java

```diff
@@ -120,6 +111,62 @@ public void initializeLogging(LogConfig config, final List<RuntimeValue<Optional
         InitialConfigurator.DELAYED_HANDLER.setHandlers(handlers.toArray(EmbeddedConfigurator.NO_HANDLERS));
     }
 
+    private static void configureSyslogHandler(LogConfig config, Map<String, CategoryConfig> categories, LogContext logContext,
+            List<LogCleanupFilterElement> filterElements, ArrayList<Handler> handlers, ErrorManager errorManager) {
+        final Handler syslogHandler = configureSyslogHandler(config.syslog, errorManager, filterElements);
+        if (syslogHandler != null) {
+            handlers.add(syslogHandler);
+        }
+        for (Entry<String, CategoryConfig> entry : categories.entrySet()) {
+            final CategoryConfig categoryConfig = entry.getValue();
+            final String name = entry.getKey();
+            final Logger categoryLogger = logContext.getLogger(name);
+            categoryLogger.setUseParentHandlers(false);
```

Comment:
A concern I have with this approach is that it will squelch all other handlers, because they do not have a hook into the category configuration, and adding such a hook will be complex.  This change completely takes over the log category so it becomes impossible to mix handlers which delegate to parents, and ones which don't.  The decision of whether to delegate to a parent handler for a given logger should lie with the user.

I think this problem might be better solved at the formatter level, by adding a mechanism for the formatter to choose a format based on factors such as category.

Reply:
I thought the setUseParentHandler() was used wether or not to take the handlers from the "root" logger. I'll give it another look, but indeed it was not my intention to break the hierarchy.

However, the suggestion you make is to add a format property in CategoryConfig instead of giving the ability to override the complete ( console/file/syslog ) handler ?
In our application for example we could write everything to file, and certain appenders only to console. If we only allow to override the format this is still an "issue" and we pollute the file as well.

Reply:
Each handler has its own formatter, so you don't have to worry about cross-pollution between them at that level.

Maybe we do need to work up a more generic configuration setup for log handlers.  We could introduce a `useParentHandler` property on the category configuration, and otherwise introduce per-handler configuration in the manner that you have done.

The problem to solve would be figuring out how to deal with pluggable formatters, since they could potentially apply to any handler on any category, and the current infrastructure doesn't account for that.

Finding a way to do this compatibly is an added challenge.